PKHeX.Core/Saves/Encryption/MemeCrypto/MemeKey.cs

```diff
@@ -22,9 +23,13 @@ namespace PKHeX.Core;
     /// <summary> Modulus, BigInteger </summary>
     private readonly BigInteger N;
 
+    private readonly IAesCryptographyProvider _aesProvider;
+
     // Constructor
-    public MemeKey(MemeKeyIndex key)
+    public MemeKey(MemeKeyIndex key, IAesCryptographyProvider? aesProvider = null)
     {
+        _aesProvider = aesProvider ?? IAesCryptographyProvider.Default;
```

Comment:
**note:** A `MemeKey` can optionally accept an implementation of the algorithm, but if none is provided, it will use the `Default`, which is backed up by the native .NET implementation through the call to: `Aes.Create()`